koin-projects/koin-core/src/main/kotlin/org/koin/core/Koin.kt

```diff
@@ -111,15 +112,19 @@ class Koin {
      * Declare a component definition from the given instance
      * This result of declaring a single definition of type T, returning the given instance
      *
-     * @param instance
-     * @param qualifier
+     * @param instance The instance you're declaring.
+     * @param qualifier Qualifier for this declaration
+     * @param secondaryTypes List of secondary bound types
+     * @param options Allows to set [Options] for the [BeanDefinition] that will be created, e.g. you can use it to
+     * override a previous declaration of the same type.
      */
     inline fun <reified T> declare(
             instance: T,
             qualifier: Qualifier? = null,
-            secondaryTypes: List<KClass<*>>? = null
+            secondaryTypes: List<KClass<*>>? = null,
+            options: Options? = null
```

Comment:
I think it's easier for the user if these `Options` were written like here: https://github.com/InsertKoinIO/koin/blob/12e06086e557cf5d33e5e5e3c7bc52c797515bf1/koin-projects/koin-core/src/main/kotlin/org/koin/dsl/Module.kt#L29

Just booleans (or whatever type), so the users have a clear and consistent API.

Reply:
I would love to know @arnaudgiuliani 's point of view. I can easily change it to have just a:

```kotlin
inline fun <reified T> declare(
            instance: T,
            qualifier: Qualifier? = null,
            secondaryTypes: List<KClass<*>>? = null,
            override: Boolean = false
```

But I thought that having `Options` as a parameter offers an easier way to extend this API in the future.  I'm fine either way.